Client/src/view/BattleView/MapBox.java

```diff
@@ -258,59 +261,92 @@ void updateMapColors() {
-
-                        } else {
-                            cells[row][column].setFill(Constants.SPELL_COLOR);
-                        }
-                    } else
-                        cells[row][column].setFill(Constants.defaultColor);
-                    continue;
+                    Boolean update = updateMapColoursOnInsertion(row, column, player);
+                    //if (update) { continue; }
                 }
-                if (selectionType == SelectionType.SELECTION) {
-                    if (currentTroop != null && currentTroop.getPlayerNumber() == battleScene.getMyPlayerNumber()) {
-                        cells[row][column].setFill(Constants.CAN_SELECT_COLOR);
-                    } else
-                        cells[row][column].setFill(Constants.defaultColor);
-                    continue;
-                }
-                if (selectedTroop != null && selectedTroop.getCell().getRow() == row &&
-                        selectedTroop.getCell().getColumn() == column) {
-                    cells[row][column].setFill(Constants.SELECTED_COLOR);//not important
-                    continue;
+                if (selectionType == SelectionType.NORMAL){
+                    boolean updateNormal = updateMapColoursOnNormal(row, column, player);
+                    if (updateNormal) { continue;}
                 }
 
-                if (selectionType == SelectionType.SPELL) {
-                    cells[row][column].setFill(Constants.defaultColor);
+                if (selectedTroop != null && selectedTroop.getCell() == new Cell(row, column)){
+                    cells[row][column].setFill(Constants.SELECTED_COLOR);
                     continue;
                 }
-                if (selectionType == SelectionType.NORMAL) {
-                    if (GameController.getInstance().getAvailableActions().canAttack(gameMap, player, selectedTroop, row, column))
-                        cells[row][column].setFill(Constants.ATTACK_COLOR);
-                    else if (GameController.getInstance().getAvailableActions().canMove(gameMap, player,
-                            selectedTroop, row, column))
-                        cells[row][column].setFill(Constants.MOVE_COLOR);
-                    else
-                        cells[row][column].setFill(Constants.defaultColor);
+
+                updateMapColourHighlightEnemyUnits(row, column, player);
+                updateMapColoursHighlightUnitActions(row, column, player);
+            }
+        }
+    }
+
+    private void updateMapColoursHighlightUnitActions(int row, int column, CompressedPlayer player){
+        Troop troop = getTroop(row, column);
+        if (troop == null){ return; }
+        if (!battleScene.isMyTurn()) { return; }
+        if (troop.getPlayerNumber() != player.getPlayerNumber()) { return;}
+
+        if (troop.canMove()){
+            cells[row][column].setFill(Constants.CAN_MOVE);
+        }
+        //if (troop.canAttack()){
```

Comment:
Should this be removed or uncommented?

Reply:
Its commented for now since long term we will want a visual indicator to not if unit can attack.

I started the work here, but decided it was better to commit what I had and worry about attack stuff later